### PR TITLE
Force `START_TEST` and `STOP_TEST` to be always used in tandem

### PR DESCRIPTION
**Pull request by zickgraf**
See commit messages for details. The third commit ("Force START_TEST and STOP_TEST to appear as a pair with the same name") changes the meaning of the first argument of `STOP_TEST`. If you do not like this (or it causes to many test failures), I will remove this commit.

## Text for release notes

TBD

### CODE REVIEW

**Comment by zickgraf** [comment]
As expected, the CI is failing, e.g. due to the `STOP_TEST` in `tst/testinstall/grp/classic-PS.tst` which is called with "classic.tst-PS" instead of "classic-PS.tst". I can fix those problems but want to make sure you like the change before investing time.

**Comment by fingolfin** [comment]
Even if you fix all those failures here, it is very likely that many similar failures will exist in packages. So it only makes sense to work on this aspect if you *also* plan to get all affected packages updated. In that case, the algorithm would be this:
1. fix the immediate issues on this PR (to make CI pass here)
2. afterwards I'll start a run of the `PackageDistro` CI using this PR, and we can collect the failing packages.
3. Then for each package a PR must be made to update it ...
4. ... and the maintainer(s) must be coaxed into releasing updates for their packages
5. We can re-do 2-4 as needed, until all passes again
6. finally merge this

File: lib/profile.g
Comment (fingolfin) [inline lib/profile.g:1114]:
```suggestion
    local time;
```

File: lib/test.gi
Comment (fingolfin) [inline lib/test.gi:870]:
```suggestion
  STOP_TEST := STOP_TEST_QUIET;
```

**Comment by ChrisJefferson** [comment]
Have you considered switching to an info message rather than `Error`? We already have a few places where we use `InfoWarning` at level 1 to print out problems with test files.

File: lib/test.gi
Comment (zickgraf) [reply lib/test.gi:870]: This is on purpose: the return value has to be dropped, otherwise it gets printed during the tests.

**Comment by zickgraf** [comment]
> Have you considered switching to an info message rather than `Error`? We already have a few places where we use `InfoWarning` at level 1 to print out problems with test files.

Thanks for the suggestion, that sounds like a good idea. I would make it print at level 2 to not have to adapt all packages, would that be okay?

**Comment by fingolfin** [comment]
@zickgraf sure

**Comment by ChrisJefferson** [comment]
Easiest to start at level 2, then move to level 1 after packages with issues have been giving a reasonable chance to fix things.

 Personally (other reviewers may disagree!) I wouldn't mind if some packages printed an info message about inconsistent END_TEST message, as it wouldn't stop those package's tests from passing (unlike an Error, which leads to the tests appearing to fail).

**Comment by zickgraf** [comment]
I have turned the errors into warnings and also fixed various test files calling STOP_TEST with a wrong identifier (or not at all).

> Personally (other reviewers may disagree!) I wouldn't mind if some packages printed an info message about inconsistent END_TEST message, as it wouldn't stop those package's tests from passing (unlike an Error, which leads to the tests appearing to fail).

The warning in START_TEST could make tests fail, so I chose info level 2 for now.

**Review by fingolfin — APPROVED** [review]

File: tst/testbugfix/2018-12-06-GroupWithGenerators.tst
Comment (fingolfin) [inline tst/testbugfix/2018-12-06-GroupWithGenerators.tst:1]: I don't understand: why is this needed?

Comment (zickgraf) [reply tst/testbugfix/2018-12-06-GroupWithGenerators.tst:1]: Otherwise, it prints a warning due to InfoPerformance not being zero.
Previously, TestDirectory did not properly call `STOP_TEST` which is why
InfoPerformance was always 0 after the first `START_TEST`.

**Comment by zickgraf** [comment]
Unexpectedly, in one test the actual output changed :o I will have to look into the details when I find the time again.

**Comment by fingolfin** [comment]
Not a surprise: IIRC `START_TEST` resets the RNG state, and so my theory would be that by inserting one `START_TEST` you modified the RNG state for all subsequent tests. And it just so happens that there is one test file without its own `START_TEST` but which is affected by the RNG state... fix should be to add `START_TEST` to that file (to prevent similar problems in the future) and then adjust the output

**Comment by zickgraf** [comment]
> Not a surprise: IIRC `START_TEST` resets the RNG state, and so my theory would be that by inserting one `START_TEST` you modified the RNG state for all subsequent tests. And it just so happens that there is one test file without its own `START_TEST` but which is affected by the RNG state... fix should be to add `START_TEST` to that file (to prevent similar problems in the future) and then adjust the output

Thanks for the hint, that indeed was the culprit! Now finally everything is green :-)

**Comment by ChrisJefferson** [comment]
Just checking, are you now happy with this PR? I am, but just wanted to check before merging.

File: lib/test.gi
Comment (fingolfin) [reply lib/test.gi:870]: Then perhaps add a comment stating that?

**Review by fingolfin — APPROVED** [review]
Thank you! One minor suggestion, but fine anyway

**Pull request merged**
